com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -454,7 +457,29 @@ public override void Setup(ScriptableRenderContext context, ref RenderingData re
 
             // The copying of depth should normally happen after rendering opaques.
             // But if we only require it for post processing or the scene camera then we do it after rendering transparent objects
-            m_CopyDepthPass.renderPassEvent = (!requiresDepthTexture && (applyPostProcessing || isSceneViewCamera || isGizmosEnabled)) ? RenderPassEvent.AfterRenderingTransparents : RenderPassEvent.AfterRenderingOpaques;
+            // Aim to have the most optimized render pass event for Depth Copy (The aim is to minimize the number of render passes)
+            if (requiresDepthTexture)
+            {
+                RenderPassEvent copyDepthPassEvent = RenderPassEvent.AfterRenderingTransparents;
+                // CameraData's requiresDepthTexture is based on the URP Pipeline Asset UI checkbox value.
+                if (cameraData.requiresDepthTexture)
+                {
+                    // If user explicitly enable the setting, we force the depth copy to be AfterRenderingSkybox
+                    copyDepthPassEvent = (RenderPassEvent)Mathf.Min((int)copyDepthPassEvent, (int)RenderPassEvent.AfterRenderingSkybox);
+                }
+                // RenderPassInputs's requiresDepthTexture is configured through ScriptableRenderPass's ConfigureInput function
+                if (renderPassInputs.requiresDepthTexture)
+                {
+                    // Do depth copy before the render pass that requires depth texture as shader read resource
+                    copyDepthPassEvent = (RenderPassEvent)Mathf.Min((int)copyDepthPassEvent, ((int)renderPassInputs.requiresDepthTextureEarliestEvent) - 1);
+                }
+                m_CopyDepthPass.renderPassEvent = copyDepthPassEvent;
+            }
+            else if (cameraHasPostProcessingWithDepth || isSceneViewCamera || isGizmosEnabled)
+            {
+                // If only post process requires depth texture, we can re-use depth buffer from main geometry pass instead of enqueuing a depth copy pass, but no proper API to do that for now, so resort to depth copy pass for now
+                m_CopyDepthPass.renderPassEvent = RenderPassEvent.AfterRenderingTransparents;
+            }
```

Comment:
Nitpick: It feels more readable to me that this logic setups the `inputSummary.requiresDepthTextureEarliestEvent`
 and `copydepthPass.rendererPassEvent = inputSummary.requiresDepthTextureEarlistEvent`

Reply:
after having a second look, I think it makes more sense the way requiresDepthTextureEarliestEvent is initialized now in GetRenderPassInputs.
Since GetRenderPassInputs is the place where all the other inputSummary fields are initialized I think could be confusing to move that logic somewhere else? (also might affect initialization orders?)